### PR TITLE
qt: Make sure the statusbar reflects internal states correct

### PR DESCRIPTION
**Pull request by xdustinface**
This PR makes sure that the statusbar label, the statusbar progressbar and the statusbar sync/connection icons do always represent the actual current networking/syncing state of the wallet. Before this PR they had literally no meaningful connection to the internals most of the time time.

The statusbar label/progressbar will always show/hide now if required and show the correct internal state.

The sync/spinner icon will now disappear if there are no connections available or if the network gets disabled by `setnetworkactive` RPC. Before it was just always there, the spinner was mostly inactive and if the wallet got a sync once, it just always showed the synced icon no matter what. As soon as the wallet gets one connection from now on the spinner will pop up start to animate to signal that the wallet is actually doing something. If its done syncing it gets the green checkmark like before. But if it gets out of sync again due to timeout or `mnsync` RPC it will directly represent this unlike before where it always showed synced. 

Also the connection icon does animate now (show/hide the lower dot in orange with 500ms cycle) if there are no connections available to signal that the wallet is trying to connect to the network.  

[Click here](https://youtu.be/whZmrQH1vsc ) for a preview video of the sync process (i cut some of the middle/governance sync part) 

Based on #3688 + #3690

### CODE REVIEW

**Comment by UdjinM6** [comment]
Needs rebase

**Comment by xdustinface** [comment]
> Needs rebase

Done

File: src/qt/bitcoingui.cpp [hunk 1]
```diff
@@ -1002,7 +1088,10 @@ void BitcoinGUI::gotoVerifyMessageTab(QString addr)
 
 void BitcoinGUI::updateNetworkState()
 {
+    static int nPreviousCount{0};
+    static bool fPreviousState{false};
     int count = clientModel->getNumConnections();
+    bool fNetworkState = clientModel->getNetworkActive();
```
Comment (UdjinM6) [inline src/qt/bitcoingui.cpp:1094]: nit: imo "state" doesn't fit `bool` very well. Maybe s/`fNetworkState`/`fNetworkActive`/ (and s/`fPreviousState`/`fNetworkActivePrev`/)?

**Review by UdjinM6 — COMMENTED** [review]
Looks good in general and seems to be working as expected. Just a few code style related suggestions.

File: src/qt/bitcoingui.cpp [hunk 2]
```diff
@@ -1119,6 +1117,20 @@ void BitcoinGUI::updateHeadersSyncProgressLabel()
         progressBarLabel->setText(tr("Syncing Headers (%1%)...").arg(QString::number(100.0 / (headersTipHeight+estHeadersLeft)*headersTipHeight, 'f', 1)));
 }
 
+void BitcoinGUI::updateProgressBarVisibility()
+{
+    if (clientModel != nullptr) {
```
[the remark this answers is not in the document]
Comment (xdustinface) [reply src/qt/bitcoingui.cpp]: Done in 9046b68

File: src/qt/bitcoingui.cpp (hunk 1 above)
Comment (xdustinface) [reply src/qt/bitcoingui.cpp]: Done in af2db17

**Review by UdjinM6 — APPROVED** [review]
ACK

File: src/qt/bitcoingui.cpp [hunk 3]
```diff
@@ -1014,15 +1103,49 @@ void BitcoinGUI::updateNetworkState()
...
         labelConnectionsIcon->setToolTip(tr("Network activity disabled"));
         icon = "connect_4";
         color = GUIUtil::ThemedColor::RED;
     }
 
-    labelConnectionsIcon->setPixmap(GUIUtil::getIcon(icon, color).pixmap(STATUSBAR_ICONSIZE, STATUSBAR_ICONSIZE));
+    if (fNetworkActive && count == 0) {
+        startConnectingAnimation();
+    }
+    if(!fNetworkActive || count > 0) {
```
Comment (PastaPastaPasta) [inline src/qt/bitcoingui.cpp:1145]:
```suggestion
    if (!fNetworkActive || count > 0) {
```

**Review by PastaPastaPasta — CHANGES_REQUESTED** [review]

**Review by PastaPastaPasta — COMMENTED** [review]
overall looks good, just that nit

**Review by UdjinM6 — APPROVED** [review]
utACK

**Review by PastaPastaPasta — APPROVED** [review]
ACK

**Pull request merged**